### PR TITLE
[Yul] Use C++ user-defined literals for creating YulString constants.

### PR DESCRIPTION
**Pull request by ekpyron**
I think this is a good middle ground between the convenience of implicit construction from strings and the verbosity of the full constructor.

@chriseth What do you think?

### CODE REVIEW

File: libyul/YulString.h [hunk 1]
```diff
@@ -127,4 +127,9 @@ class YulString
 	YulStringRepository::Handle m_handle{ 0, YulStringRepository::emptyHash() };
 };
 
+inline YulString operator "" _yul(const char* _string, std::size_t _size)
```
Comment (ekpyron) [inline libyul/YulString.h]: We could also use a different suffix, e.g  ``_ys``.

**Comment by chriseth** [comment]
While this looks really nice, I'm not sure whether it would be more confusing than useful, especially given that such constructions are not used that often.

Would like to hear other opinions, though.

**Comment by ekpyron** [comment]
It does not only look nice, but moving forward it can potentially have performance benefits:
Once we move to C++14 we can change the user-defined literal operator to a templated version (the signature would be ``template<typename CH, CH... chars> YulString operator "" _yul()``), which we could use to compute the string hash at compile time. Not that this would make a huge difference, but still.
And I don't think it'd be all that confusing, but we can collect opinions on that, I'm requesting more reviews.

**Comment by leonardoalt** [comment]
In my opinion, if it has no performance benefits I'd stick to the current one. But if we do get performance improvements when C++14 I'd vote for merging.

**Review by chriseth — APPROVED** [review]

**Pull request merged**